pulsar-broker/src/main/java/org/apache/pulsar/broker/service/Producer.java

```diff
@@ -23,6 +23,9 @@
 import static org.apache.pulsar.common.api.Commands.hasChecksum;
 import static org.apache.pulsar.common.api.Commands.readChecksum;
 
+import java.util.Collections;
+import java.util.HashMap;
```

Comment:
nit: `HashMap` is probably redundant now.

Reply:
fixing.